tsconfig.json

```diff
@@ -8,7 +8,7 @@
     "moduleResolution": "node",
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
-    "noLib": true,
+    //"noLib": true,
```

Comment:
cc @kitsonk

Reply:
Shouldn't need this.  @piscisaureus what type errors were we getting, there must be something "missing" from the libs we want/need to use.  It will potentially mask other problems if we eliminate this.

Reply:
I pulled this branch and reverted to `"noLib": true` and didn't get any issues with `./tools/build.py :main_bundle` or `./tools/build.py :compiler_bundle`.

Reply:
I commented it out because I use VS code to hack deno itself.
With `noLib` enabled intellisense doesn't work at all (e.g. it doesn't even know about `Math`).

Reply:
Hmmm... it _should_ pick up the lib from the `files` array, but I have noticed with TypeScript 3.3 which is now default with the latest version of VSCode that it doesn't do this and I don't know why.  I use the project specific version of TypeScript when editing Deno:

<img width="604" alt="metrics_ts_ _deno" src="https://user-images.githubusercontent.com/1282577/52754761-cd836200-304f-11e9-9d50-4c30666b3b90.png">

And I don't have the problem.

When I tried to upgrade Deno to TS3.3 I got the same problem and quickly looked at the issues and didn't see anyone else reporting it, so I ignored it and moved on to other issues.  I clearly need to try again and narrow down where the problem is coming from.

Reply:
@kitsonk switching to TS workspace version does fix the vscode issue for me, so I've reverted the noLib change on this branch.
That said, great if you look into making TS 3.3 work.